core/src/main/scala/org/apache/spark/api/r/RBackendHandler.scala

```diff
@@ -143,12 +142,8 @@ private[r] class RBackendHandler(server: RBackend)
       val cls = if (isStatic) {
         Utils.classForName(objId)
       } else {
-        JVMObjectTracker.get(objId) match {
-          case None => throw new IllegalArgumentException("Object not found " + objId)
-          case Some(o) =>
-            obj = o
-            o.getClass
-        }
+        obj = server.jvmObjectTracker.get(JVMObjectId(objId))
```

Comment:
+1 for throwing an exception with old error message. Or even better add more description. The R users generally complain about cryptic error messages in SparkR.